types/src/client.rs

```diff
@@ -109,63 +117,31 @@ where
 	Notif: DeserializeOwned,
 {
 	/// Returns the next notification from the stream
-	/// This may return `None` if the subscription has been terminated,
-	/// may happen if the channel becomes full or is dropped.
-	///
-	/// Ignores any malformed packet.
-	pub async fn next(&mut self) -> Option<Notif> {
-		loop {
-			match self.notifs_rx.next().await {
-				Some(n) => match serde_json::from_value(n) {
-					Ok(parsed) => return Some(parsed),
-					Err(e) => {
-						log::error!("Subscription response error: {:?}", e);
-					}
-				},
-				None => return None,
-			}
-		}
-	}
-}
-
-impl<Notif> NotificationHandler<Notif>
-where
-	Notif: DeserializeOwned,
-{
-	/// Returns the next notification from the stream
-	/// This may return `None` if the method has been unregistered,
+	/// This may return `Ok(None)` if the subscription has been terminated,
 	/// may happen if the channel becomes full or is dropped.
```

Comment:
```suggestion
	/// Returns `Ok(None)` if the subscription has been terminated on the server side, e.g. the server dropped the
	/// connection, crashed, or dropped the connection for whatever reason.
```

I am not sure I understand how the channel being full would cause `None`. Can you help me understand? Wouldn't that mean that the server can't send messages on it but the client could still read from it?